hummingbot/strategy/avellaneda_market_making/start.py

```diff
@@ -17,28 +23,35 @@
 from hummingbot.strategy.avellaneda_market_making import (
     AvellanedaMarketMakingStrategy,
 )
-from hummingbot.strategy.avellaneda_market_making.avellaneda_market_making_config_map import avellaneda_market_making_config_map as c_map
 
 
 def start(self):
     try:
-        order_amount = c_map.get("order_amount").value
-        order_optimization_enabled = c_map.get("order_optimization_enabled").value
-        order_refresh_time = c_map.get("order_refresh_time").value
-        exchange = c_map.get("exchange").value.lower()
-        raw_trading_pair = c_map.get("market").value
-        max_order_age = c_map.get("max_order_age").value
-        inventory_target_base_pct = 0 if c_map.get("inventory_target_base_pct").value is None else \
-            c_map.get("inventory_target_base_pct").value / Decimal('100')
-        filled_order_delay = c_map.get("filled_order_delay").value
-        order_refresh_tolerance_pct = c_map.get("order_refresh_tolerance_pct").value / Decimal('100')
-        order_levels = c_map.get("order_levels").value
-        level_distances = c_map.get("level_distances").value
-        order_override = c_map.get("order_override").value
-        hanging_orders_enabled = c_map.get("hanging_orders_enabled").value
-
-        hanging_orders_cancel_pct = c_map.get("hanging_orders_cancel_pct").value / Decimal('100')
-        add_transaction_costs_to_orders = c_map.get("add_transaction_costs").value
+        c_map: AvellanedaMarketMakingConfigMap = self.strategy_config_map
+        order_amount = c_map.order_amount
+        order_optimization_enabled = c_map.order_optimization_enabled
+        order_refresh_time = c_map.order_refresh_time
+        exchange = c_map.exchange
+        raw_trading_pair = c_map.market
+        max_order_age = c_map.max_order_age
+        inventory_target_base_pct = 0 if c_map.inventory_target_base_pct is None else \
+            c_map.inventory_target_base_pct / Decimal('100')
+        filled_order_delay = c_map.filled_order_delay
+        order_refresh_tolerance_pct = c_map.order_refresh_tolerance_pct / Decimal('100')
+        if isinstance(c_map.order_levels_mode, MultiOrderLevelModel):
+            order_levels = c_map.order_levels_mode.order_levels
+            level_distances = c_map.order_levels_mode.level_distances
+        else:
+            order_levels = 1
+            level_distances = 0
```

Comment:
Could we request the required information to the order levels mode in a polymorphic way? That would permit us to avoid checking the type of the object (and would make the solution more extensible)

Reply:
Same as above